Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1909 @@
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: bind( {}, ({},{:08x}:{}), {} ):{} ({})\n", fd,
+                       addr.sin_family, addr.sin_addr.s_addr, Common::swap16(addr.sin_port), len,
+                       ret, err);
+
+        s_media_buffer_32[1] = ret;
+        s_last_error = SSC_SUCCESS;
+        break;
+      }
+      case AMMBCommand::Closesocket:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+
+        const int ret = closesocket(fd);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: closesocket( {}({}) ):{}\n", fd,
+                       s_media_buffer_32[2], ret);
+
+        s_sockets[SocketCheck(s_media_buffer_32[2])] = SOCKET_ERROR;
+
+        s_media_buffer_32[1] = ret;
+        s_last_error = SSC_SUCCESS;
+        break;
+      }
+      case AMMBCommand::Connect:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+        const u32 off = s_media_buffer_32[3] - NetworkCommandAddress2;
+        const u32 len = s_media_buffer_32[4];
+
+        if (!NetworkCMDBufferCheck(off))
+        {
+          break;
+        }
+
+        sockaddr_in addr;
+        memcpy(&addr, s_network_command_buffer + off, sizeof(sockaddr_in));
+
+        const int ret = NetDIMMConnect(fd, &addr, len);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: connect( {}({}), ({},{}:{}), {} ):{}\n", fd,
+                       s_media_buffer_32[2], addr.sin_family, inet_ntoa(addr.sin_addr),
+                       Common::swap16(addr.sin_port), len, ret);
+
+        s_media_buffer[1] = s_media_buffer[8];
+        s_media_buffer_32[1] = ret;
+        break;
+      }
+      case AMMBCommand::InetAddr:
+      {
+        const char* ip_address = reinterpret_cast<char*>(s_network_command_buffer);
+
+        // IP address shouldn't be longer than 15
+        if (strnlen(ip_address, 15) > 15)
+        {
+          ERROR_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: Invalid size for address: InetAddr():{}\n",
+                        strlen(ip_address));
+          break;
+        }
+
+        const u32 ip = inet_addr(ip_address);
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: InetAddr( {} )\n", ip_address);
+
+        s_media_buffer[1] = s_media_buffer[8];
+        s_media_buffer_32[1] = Common::swap32(ip);
+        break;
+      }
+      case AMMBCommand::Listen:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+        const u32 backlog = s_media_buffer_32[3];
+
+        const int ret = listen(fd, backlog);
+
+        NOTICE_LOG_FMT(AMMEDIABOARD_NET, "GC-AM: listen( {}, {} ):{:d}\n", fd, backlog, ret);
+
+        s_media_buffer[1] = s_media_buffer[8];
+        s_media_buffer_32[1] = ret;
+        break;
+      }
+      case AMMBCommand::Recv:
+      {
+        const SOCKET fd = s_sockets[SocketCheck(s_media_buffer_32[2])];
+        const u32 off = s_media_buffer_32[3];
+        const auto len = std::min<u32>(s_media_buffer_32[4], sizeof(s_network_buffer));
```

Comment:
This needs to take into account that the maximum allowed length is smaller if `off` is larger than `NetworkBufferAddress4`.